bundles/specmate-model-generation/src/com/specmate/modelgeneration/GenerateModelFromPseudoCode.java

```diff
@@ -0,0 +1,370 @@
+package com.specmate.modelgeneration;
+
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.emf.common.util.URI;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+
+import com.specmate.common.exception.SpecmateException;
+import com.specmate.common.exception.SpecmateInternalException;
+import com.specmate.model.administration.ErrorCode;
+import com.specmate.model.requirements.CEGModel;
+import com.specmate.model.requirements.CEGNode;
+import com.specmate.model.requirements.NodeType;
+import com.specmate.objectif.resolve.BusinessRuleUtil;
+import com.specmate.objectif.resolve.rule.AndNode;
+import com.specmate.objectif.resolve.rule.BusinessRuleNode;
+import com.specmate.objectif.resolve.rule.LiteralNode;
+import com.specmate.objectif.resolve.rule.ObjectifNode;
+import com.specmate.objectif.resolve.rule.OrNode;
+import com.specmate.xtext.XTextException;
+
+// This class transforms pseudo code into a CEG.
+// The pseudo code must conform with the implemented XText grammar.
+public class GenerateModelFromPseudoCode implements ICEGFromRequirementGenerator {
+
+	// Our factory to create the elements of the CEG (nodes and connections).
+	CEGCreation creation = new CEGCreation();
+
+	// Counter for numbering of intermediate nodes as well as business rule nodes.
+	int intermediatenum = 1;
+	int brnum = 1;
+
+	@Override
+	public CEGModel createModel(CEGModel model, String text) throws SpecmateException {
+		List<BusinessRuleNode> rules;
+		// First, we need to load all pseudo code lines from a document / input.
+		try {
+			rules = new BusinessRuleUtil().parseXTextResource(text);
+		} catch (XTextException e) {
+			throw new SpecmateInternalException(ErrorCode.INTERNAL_PROBLEM, "...");
```

Comment:
please add a sensible exception message. something that indicates that the xtext pasrsing went wrong. also include the orginal message (e.getMessage())

Reply:
fixed in new commit